libs/MeshKernel/src/Mesh2D.cpp

```diff
@@ -1934,3 +1936,167 @@ std::vector<int> Mesh2D::NodeMaskFromPolygon(const Polygons& polygon, bool insid
+{
+    // Use a flood fill like algorithm to label all elements, connected via edges, with the same region id.
+
+    std::queue<UInt> toBeProcessed;
+    // Add a single unlabeled element to the queue
+    toBeProcessed.push(unlabledElementId);
+    elementRegionId[unlabledElementId] = regionId;
+    elementCount = 1;
+
+    // Process the queue, until there are no nodes left to process.
+    while (!toBeProcessed.empty())
+    {
+        UInt currentElement = toBeProcessed.front();
+        toBeProcessed.pop();
+
+        // Get the neighbours of this node from the element-edge connectivity graph.
+        // Only across faces
+        const std::vector<UInt>& edges = m_facesEdges[currentElement];
+
+        for (UInt i = 0; i < edges.size(); ++i)
+        {
+            UInt neighbour = GetNeighbour(m_edgesFaces[edges[i]], currentElement);
+
+            if (neighbour != constants::missing::uintValue && elementRegionId[neighbour] == constants::missing::uintValue)
+            {
+                // If any neighbour element is labeled undefined it will get current region id
+                // and added to the end of the queue to be processed later.
+                elementRegionId[neighbour] = regionId;
+                ++elementCount;
+                toBeProcessed.push(neighbour);
+            }
+        }
+    }
+}
+
+void Mesh2D::LabelSingleDomainRegion(const UInt regionId, std::vector<UInt>& elementRegionId, UInt& elementCount) const
+{
+    UInt unlabledElementId = constants::missing::uintValue;
+    elementCount = 0;
+
+    // Find an element that has not yet been labeled, any as yet unlabeled element will do,
+    // so use the first one that we find in the list.
+    for (UInt i = 0; i < elementRegionId.size(); ++i)
+    {
+        // Find first unlabeled element
+        if (elementRegionId[i] == constants::missing::uintValue)
+        {
+            unlabledElementId = i;
+            break;
+        }
+    }
+
+    // If such an element is found then label all connected elements (across faces)
+    // with the current regionId.
+    if (unlabledElementId != constants::missing::uintValue)
+    {
+        LabelConnectedRegion(regionId, elementRegionId, unlabledElementId, elementCount);
+    }
+}
+
+void Mesh2D::LabelAllDomainRegions(std::vector<UInt>& elementRegionId, std::vector<std::pair<UInt, UInt>>& regionCount) const
+{
+    bool allRegionsLabeled = false;
+    UInt regionId = 1;
+
+    elementRegionId.resize(GetNumFaces(), constants::missing::uintValue);
+    regionCount.clear();
+
+    while (!allRegionsLabeled)
+    {
+        UInt elementCount = 0;
+        LabelSingleDomainRegion(regionId, elementRegionId, elementCount);
+
+        if (elementCount > 0)
+        {
+            // If more than 0 elements were labaled, then save this region-id and number of elements.
+            regionCount.push_back({regionId, elementCount});
```

Comment:
can emplace back the pair: `regionCount.emplace_back(regionId, elementCount);`

Reply:
Done